src/core/mesh/qgsmeshdatasetgroupstore.h

```diff
@@ -127,8 +127,8 @@ class QgsMeshDatasetGroupStore: public QObject
     //! Constructor
     QgsMeshDatasetGroupStore( QgsMeshLayer *layer );
 
-    //! Sets the persistent mesh data provider
-    void setPersistentProvider( QgsMeshDataProvider *provider );
+    //!  Sets the persistent mesh data provider with the path of its extra dataset
```

Comment:
you can have multiple extra datasets, right?

Reply:
yes

Reply:
my question was more how it is possible to add multiple extra datasets with this interface?

Reply:
`setPersistentProvider( QgsMeshDataProvider *provider, const QStringList &extraDatasetUri )` is called only once time, after the provider is created and it is valid. 
Two cases : 
1- when the mesh is loaded for the first time (edited) 
2-when the mesh layer is loaded from a project, extraDatasetUri contains the user extra datasets